### PR TITLE
Fix warnings reported by api-review for AAD and AAD-B2C

### PR DESCRIPTION
**Pull request by chenrujun**
1. Reorganize the package of AAD and AAD B2C related classes.
      - AAD and AADB2C are different thing, not put AAD B2C related classes into AAD package.
      - Move configuration classes and property classes out of "implementation" package.
      - Move Classes used by customer out of "implementation" package.
      - If a class is used in a non-implementation class's public method's parameter type / return type. Move it out of "implementation" package.
2. Add missed java docs.

Here is the link to api-review after this PR: https://apiview.dev/Assemblies/Review/627f1d980d854f0fb219ad4d9a48e54c

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/properties/AADProfileProperties.java [hunk 1]
```diff
@@ -23,27 +23,51 @@
...
+     *
+     * @return The clout type.
+     */
     public AzureProfileAware.CloudType getCloud() {
         return cloud;
     }
 
+    /**
+     *
+     * @param cloud The clout type.
```
Comment (stliu) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/properties/AADProfileProperties.java:52]: typo

Comment (chenrujun) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/properties/AADProfileProperties.java]: Fixed.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aadb2c/configuration/auto/AADB2CAutoConfiguration.java [hunk 2]
```diff
@@ -1,8 +1,14 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
-package com.azure.spring.cloud.autoconfigure.aad.b2c.implementation;
+package com.azure.spring.cloud.autoconfigure.aadb2c.configuration.auto;
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aadb2c/configuration/auto/AADB2CAutoConfiguration.java:3]: same here

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/configuration/auto/AADAuthenticationFilterAutoConfiguration.java [hunk 3]
```diff
@@ -1,11 +1,14 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-package com.azure.spring.cloud.autoconfigure.aad.filter;
+package com.azure.spring.cloud.autoconfigure.aad.configuration.auto;
```
[the remark this answers is not in the document]
Comment (chenrujun) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/configuration/auto/AADAuthenticationFilterAutoConfiguration.java]: Updated.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aadb2c/configuration/auto/AADB2CAutoConfiguration.java (hunk 2 above)
Comment (chenrujun) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aadb2c/configuration/auto/AADB2CAutoConfiguration.java]: Updated.

**Review by saragluna — APPROVED** [review]
LGTM

**Comment by chenrujun** [comment]
Merging this PR. 
Ignoring the pipeline because the failure is not caused by this PR.

**Pull request merged**